windows/mdns.h

```diff
@@ -0,0 +1,1535 @@
+/* mdns.h  -  mDNS/DNS-SD library  -  Public Domain  -  2017 Mattias Jansson
```

Comment:
I'm not familiar with Windows development, but why are we including a library inside the plugin codebase?
I'd prefer that we either depend on official Windows API or use a dependency management system to depend on this library.

Reply:
Maybe via [Conan](https://conan.io)?

Reply:
> Maybe via [Conan](https://conan.io)? I see it is published there https://github.com/mjansson/mdns#using-with-cmake-conan-or-vcpkg

Unfortunately, windows has no built-in support for mdns. So we have to rely on 3rd party libraries. There are several available. I chose this one because it was very self-contained (just a single .h file) and seemed relatively stable.

I agree it would be better to include a dependency rather than copy/paste. I just did not know how to accomplish that so I wanted to see if it would work in the first place.